### PR TITLE
Hadoop 16687: Fix testcase added for HADOOP-16138 for namespace enabled account

### PR DESCRIPTION
**Pull request by snvijaya**
Fixing the testcase by allowing the setup and teardown to handle creation and deletion of filesystems as expected. Moving the config to prevent filesystem creation inside the testcase which issues request to a non-existing filesystem.

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 90 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1270 | trunk passed |
| +1 | compile | 27 | trunk passed |
| +1 | checkstyle | 22 | trunk passed |
| +1 | mvnsite | 32 | trunk passed |
| +1 | shadedclient | 871 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 23 | trunk passed |
| 0 | spotbugs | 54 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 50 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 27 | the patch passed |
| +1 | compile | 22 | the patch passed |
| +1 | javac | 22 | the patch passed |
| -0 | checkstyle | 16 | hadoop-tools/hadoop-azure: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 26 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 898 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 20 | the patch passed |
| +1 | findbugs | 53 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 77 | hadoop-azure in the patch passed. |
| +1 | asflicense | 29 | The patch does not generate ASF License warnings. |
| | | 3658 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1701/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1701 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 9548095fb8b9 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / dd90025 |
| Default Java | 1.8.0_222 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1701/1/artifact/out/diff-checkstyle-hadoop-tools_hadoop-azure.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1701/1/testReport/ |
| Max. process+thread count | 308 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1701/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCLI.java [hunk 1]
```diff
@@ -57,8 +55,14 @@ public void testMkdirRootNonExistentContainer() throws Exception {
 
     String nonExistentContainer = "nonexistent-" + UUID.randomUUID();
 
+    // Disable creation of new filesystem
+    final AbfsConfiguration conf = getConfiguration();
+    conf.setBoolean(AZURE_CREATE_REMOTE_FILESYSTEM_DURING_INITIALIZATION,
+        false);
```
Comment (iwasakims) [inline hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCLI.java]: Creating copy of Configuration (like `new Configuration(getRawConfiguration())`) would be better than updating existing AbfsConfiguration . The FsShell could be created using the updated copy in that case.

File: hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCLI.java [hunk 2]
```diff
@@ -57,8 +55,14 @@ public void testMkdirRootNonExistentContainer() throws Exception {
...
+
     int result = fsShell.run(new String[] { "-mkdir",
-        ABFS_SCHEME + "://" + nonExistentContainer + "@" + account + "/" });
+        ABFS_SCHEME + "://" + nonExistentContainer + "@" + account +
+            "/testDirOnRoot" });
```
Comment (iwasakims) [inline hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCLI.java]: Do we need this change? HADOOP-16138 looks addressing the issue of "hadoop fs mkdir /" on nonexistent container.

File: hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCLI.java (hunk 1 above)
Comment (snvijaya) [reply hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCLI.java]: Updated the rawConfiguration as suggested.

File: hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCLI.java (hunk 2 above)
Comment (snvijaya) [reply hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCLI.java]: Reverted the path back to container root.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 2078 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1237 | trunk passed |
| +1 | compile | 26 | trunk passed |
| +1 | checkstyle | 20 | trunk passed |
| +1 | mvnsite | 29 | trunk passed |
| +1 | shadedclient | 871 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 23 | trunk passed |
| 0 | spotbugs | 51 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 48 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 25 | the patch passed |
| +1 | compile | 21 | the patch passed |
| +1 | javac | 21 | the patch passed |
| +1 | checkstyle | 14 | the patch passed |
| +1 | mvnsite | 25 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 946 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 24 | the patch passed |
| +1 | findbugs | 64 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 81 | hadoop-azure in the patch passed. |
| +1 | asflicense | 36 | The patch does not generate ASF License warnings. |
| | | 5678 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1701/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1701 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux d1494dacaf3e 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 96c4520 |
| Default Java | 1.8.0_222 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1701/2/testReport/ |
| Max. process+thread count | 307 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1701/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by iwasakims** [comment]
Thanks for the update. +1. Tested on Japan East.

**Comment by snvijaya** [comment]
Thanks for commit and testing @iwasakims

**Pull request merged**